app/stac_api/models.py

```diff
@@ -177,15 +177,16 @@ def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asse
             #     asset_eo_gsd
             # )
 
-            if asset_geoadmin_variant not in self.summaries["geoadmin:variant"]:
+            if asset_geoadmin_variant and \
```

Comment:
shouldn't the function arguments be declared as optional, e.g. asset_geoadmin_variant=None, in order to do this check?

Reply:
If we leave the function signature and how we call it then you have to do it like this. Other option (probably cleaner) would be to separate these checks/updates into individual functions (since we'll probably have many use cases where not all of these properties are used). What do you think?

Reply:
You mean we'd have three separate functions update_geoadmin_variant(), update_proj_epsg() and update_eo_gsd?
Probably there is not such a big difference w.r.t. potential for mistakes.
You could either forget one or two of the optional arguments when calling the current function or you could forget to call one or two of the separate functions.

I like the way you suggest with optional arguments in one function only. But I'd rename this function to "update_summaries" instead of "update_geoadmin_variants", as geoadmin_variant is only one of the three arguments to be updated.